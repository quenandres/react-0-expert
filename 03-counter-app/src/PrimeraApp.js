import React from 'react';
//import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
// Functional components

const PrimeraApp = ({ saludo, subtitulo }) => {

    return (
        <>
            <h1> { saludo } !!</h1>
            {/* <pre>{ JSON.stringify(saludo, null, 3) }</pre> */}
            <p> { subtitulo } </p>
            <p>ad aasa 1 asasas asdasdad</p>
        </>
    );
}

// Tipos de las propiedades que va a recibir el componente
PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subtitulo ahhh'
}

export default PrimeraApp;