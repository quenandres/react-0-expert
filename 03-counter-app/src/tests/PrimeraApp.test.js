import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import PrimeraApp from '../PrimeraApp';


describe('Pruebas en <PrimeraApp />', () => {
    /* test('debe de mostrar el mensaje "Hola, Soy Goku"', () => {
        const saludo = 'Hola, soy Goku';
        const { getByText } = render(<PrimeraApp saludo={ saludo }/>); //render es una funcion que recibe un componente que se quiere renderizar
        expect( getByText(saludo) ).toBeInTheDocument();
    });   */

    test('debe de mostrar <PrimeraApp /> correctamente', () => {
        const saludo = 'Hola, soy Goku';
        const wrapper = shallow( <PrimeraApp saludo={ saludo }/> ); // Parecido a render pero puede simular click y trabajar con queryselector

        expect(wrapper).toMatchSnapshot();
    })
    

})
