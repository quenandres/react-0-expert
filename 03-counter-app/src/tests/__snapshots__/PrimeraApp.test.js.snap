// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola, soy Goku
     !!
  </h1>
  <p>
     
    Soy un subtitulo ahhh
     
  </p>
  <p>
    ad aasa 1 asasas asdasdad
  </p>
</Fragment>
`;
